src/sycl/chunked_prefill.cpp

```diff
@@ -790,81 +802,177 @@ std::vector<at::Tensor> mha_fwd(
   if (params.is_causal) {
     switch (params.d) {
       case 64:
-        FMHAConfig<
-            true,
-            cute::Shape<_128, _64, _64>,
-            cute::Shape<_128, _32, _64>,
-            cute::Shape<_128, _64, _64>,
-            cute::Layout<cute::Shape<_8, _1, _1>, cute::Stride<_1, _1, _1>>,
-            PipelineStages>::run(params);
+        if (use_sink) {
+          FMHAConfig<
+              true,
+              cute::Shape<_128, _64, _64>,
+              cute::Shape<_128, _32, _64>,
+              cute::Shape<_128, _64, _64>,
+              cute::Layout<cute::Shape<_8, _1, _1>, cute::Stride<_1, _1, _1>>,
+              PipelineStages,
+              true>::run(params);
+        } else {
+          FMHAConfig<
+              true,
+              cute::Shape<_128, _64, _64>,
+              cute::Shape<_128, _32, _64>,
+              cute::Shape<_128, _64, _64>,
+              cute::Layout<cute::Shape<_8, _1, _1>, cute::Stride<_1, _1, _1>>,
+              PipelineStages,
+              false>::run(params);
+        }
```

Comment:
use a bool dispatch `AT_DISPATCH_BOOL` to simplify the code:

```
#define AT_DISPATCH_BOOL(BOOL_V, BOOL_NAME, ...) \
  [&] {                                          \
    if (BOOL_V) {                                \
      constexpr bool BOOL_NAME = true;           \
      return __VA_ARGS__();                      \
    } else {                                     \
      constexpr bool BOOL_NAME = false;          \
      return __VA_ARGS__();                      \
    }                                            \
  }()
```

example: https://github.com/mingfeima/sglang/blob/pr_qwen3_next_support/sgl-kernel/csrc/cpu/mamba/conv.cpp#L319

Reply:
https://github.com/sgl-project/sgl-kernel-xpu/pull/19/commits/05b0ce9cb2c5c9d9753cf0a198658b199a59f04a